pkg/build/api/types.go

```diff
@@ -476,3 +479,12 @@ type BuildLogOptions struct {
 	// is not available yet. Otherwise the server will wait until the build has started.
 	NoWait bool
 }
+
+// SecretSpec specifies a secret to be included in a build pod and its corresponding mount point
+type SecretSpec struct {
```

Comment:
Why is that useful for the user?  I thought I would want to mount secrets in my actual build?

Also, this seems like an injection vector - I can bind mount a secret on top of a file that STI uses and thus get root access in a pod via a secret of a certain form...